Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp

```diff
@@ -982,7 +982,8 @@ namespace Tensile
 
                 virtual bool operator()(ContractionProblem const& problem) const override
                 {
-                    return problem.getPersistentKernelEligibility();
+                    // if value = false -> turn off check, always return eligibility as true
+                    return value ? problem.getPersistentKernelEligibility() : true;
```

Comment:
This change I would revert. A predicate should only be attached to a solution if necessary.

In the case that value = false, (don't care about PK eligibility) the predicate should not be attached anyway because it always returns true. Additional predictable check that costs time.

Only in the case that value = true (we care about PK eligibility) it should be attached to always check the problem. Not easily predictable but necessary time cost.

In fact, I don't think this predicate should have a value at all since this would indicate the same as attached or not.

Reply:
Reply together below (Contractions.py)